redisq/queue.go

```diff
@@ -235,6 +235,13 @@ func (q *Queue) Release(msg *taskq.Message) error {
 
 // Delete deletes the message from the queue.
 func (q *Queue) Delete(msg *taskq.Message) error {
+	if msg.Delay > 0 {
```

Comment:
This looks awkward (how do you know when to use Delay?) and fragile if message is moved from Z to X while you are trying to delete it.